test/addons/not-a-binding/test.js

```diff
@@ -0,0 +1,6 @@
+'use strict';
+const common = require('../../common');
+const assert = require('assert');
+
+const re = new RegExp('^Error: Module did not self-register\\.$');
+assert.throws(() => require(`./build/${common.buildType}/binding`), re);
```

Comment:
Out of curiosity, what is the reason for using a regexp object instead of a regexp literal?